packages/client/src/views/Settings/AvatarChangeModal.tsx

```diff
@@ -183,10 +184,10 @@ function AvatarChangeModalComp({
         <TertiaryButton key="cancel" id="modal_cancel" onClick={handleCancel}>
           {intl.formatMessage(buttonMessages.cancel)}
         </TertiaryButton>,
-        <Mutation<{}, { userId: string; avatar: IImage }>
+        <Mutation<{ changeAvatar: IImage }, { userId: string; avatar: IImage }>
           mutation={changeAvatarMutation}
-          onCompleted={(_) => {
-            avatar && onAvatarChanged(avatar)
+          onCompleted={({ changeAvatar: avatar }) => {
+            onAvatarChanged(avatar)
```

Comment:
I wonder if the previous approach was better when the cropped image was being saved in the component's state. Any thoughts?

Reply:
In my opinion, the latter is better. Because there might have been a sync issue; i.e. if in case the avatar change fails due to some network error,  the state should still show the updated picture which could have been misleading.